tty.h

```diff
@@ -15,6 +15,7 @@ struct tty_opts {
 	const char *ptmx;
 	struct termios termios;
 	struct termios neg_termios;
+	bool *drain;
```

Comment:
Can't we make this `bool drain` and default this to true? This would avoid a malloc.

Reply:
Yep, will do.